src/Enable.Extensions.Queuing.AzureServiceBus/Internal/AzureServiceBusQueueClient.cs

```diff
@@ -123,19 +130,36 @@ protected override void Dispose(bool disposing)
             {
                 if (disposing)
                 {
-                    _messageReceiver.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
-
-                    _messageSender.CloseAsync()
-                        .GetAwaiter()
-                        .GetResult();
+                    // Use Monitor to prevent race conditions
+                    Monitor.Enter(_queues);
+                    try
+                    {
+                        var refCount = _queue.RemoveReference();
+                        if (refCount == 0)
+                        {
+                            var removed = _queues.TryRemove(_queueKey, out _);
+                            if (removed)
```

Comment:
Let's have a new line above this `if` block.

Reply:
Done.